src/modules/hub.bicep

```
@@ -131,6 +145,207 @@ resource uploadSettingsJson 'Microsoft.Resources/deploymentScripts@2020-10-01' =
+      url: reference('Microsoft.Storage/storageAccounts/${storageAccountName}', '2019-04-01').primaryEndpoints.dfs
+      accountKey: {
+        type: 'AzureKeyVaultSecret'
+        store: {
+          referenceName: keyVaultName
+          type: 'LinkedServiceReference'
+        }
+        secretName: storageAccountName
+      }
+    }
+  }
+}
+
+module msexports 'datasets/adlsv2.bicep' = {
+  name: 'msexports'
+  dependsOn: [
+    linkedService_storageAccount
+    linkedService_keyVault
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    linkedServiceName: storageAccountName
+    compressionCodec: 'none'
+    datasetType: 'DelimitedText'
+    containerName: exportContainer.name
+  }
+}
+
+module ingestion_csv 'datasets/adlsv2.bicep' = {
+  name: 'ingestion_csv'
+  dependsOn: [
+    linkedService_storageAccount
+    linkedService_keyVault
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    linkedServiceName: storageAccountName
+    compressionCodec: 'gzip'
+    datasetType: 'DelimitedText'
+    containerName: ingestionContainer.name
+  }
+}
+
+module ingestion_parquet 'datasets/adlsv2.bicep' = {
+  name: 'ingestion_parquet'
+  dependsOn: [
+    linkedService_storageAccount
+    linkedService_keyVault
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    linkedServiceName: storageAccountName
+    compressionCodec: 'gzip'
+    datasetType: 'Parquet'
+    containerName: ingestionContainer.name
+  }
+}
+
+module transform_parquet 'pipelines/msexports_transform.bicep' = {
+  name: 'msexports_transform_parquet'
+  dependsOn: [
+    msexports
+    ingestion_parquet
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    sinkDataset:  replace('${ingestionContainer.name}_Parquet', '-', '_')
+    sourceDataset:  replace('${exportContainer.name}_DelimitedText', '-', '_')
+    ingestionContainerName: ingestionContainer.name
+    exportContainerName: exportContainer.name
+  }
+}
+
+module transform_csv 'pipelines/msexports_transform.bicep' = {
+  name: 'msexports_transform_csv'
+  dependsOn: [
+    msexports
+    ingestion_csv
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    sinkDataset:  replace('${ingestionContainer.name}_DelimitedText', '-', '_')
+    sourceDataset:  replace('${exportContainer.name}_DelimitedText', '-', '_')
+    ingestionContainerName: ingestionContainer.name
+    exportContainerName: exportContainer.name
+  }
+}
+
+module extract_parquet 'pipelines/msexports_extract.bicep' = {
+  name: 'msexports_extract_parquet'
+  dependsOn: [
+    transform_parquet
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    exportContainerName: exportContainer.name
+  }
+}
+
+module extract_csv 'pipelines/msexports_extract.bicep' = {
+  name: 'msexports_extract_csv'
+  dependsOn: [
+    transform_parquet
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    exportContainerName: exportContainer.name
+  }
+}
+
+module trigger 'triggers/adlsv2.bicep' = {
+  name: 'trigger'
```

I suspect we'll have a separate trigger for ingestion. Should we make this `trigger_msexports`?
```suggestion
  name: 'trigger_msexports'
```

---

Done